components/quickbooks/common/utils.mjs

```
@@ -44,3 +44,137 @@ export function parseLineItems(arr) {
+    throw new ConfigurationError("numLineItems must be a positive integer");
+  }
+
+  // Validate context parameter
+  if (!context || typeof context !== "object" || Array.isArray(context)) {
+    throw new ConfigurationError("context must be an object");
+  }
+
+  // Validate required keys exist for each line item
+  const missingKeys = [];
+  for (let i = 1; i <= numLineItems; i++) {
+    if (!context.hasOwnProperty(`amount_${i}`)) {
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)